server/src/main/java/io/kroki/server/service/Plantuml.java

```diff
@@ -65,8 +92,26 @@ public String decode(String encoded) throws DecodeException {
       }
     };
     this.diagramResponse = new DiagramResponse(new Caching(Version.etag()));
-    // Disable include for security reasons
-    OptionFlags.ALLOW_INCLUDE = false;
+    this.includeWhitelist = parseIncludeWhitelist(config.getString("KROKI_PLANTUML_INCLUDE_WHITELIST"));
+    // Disable unsafe include for security reasons
+    OptionFlags.ALLOW_INCLUDE = config.getBoolean("KROKI_PLANTUML_ALLOW_INCLUDE", false);
+    String plantUmlIncludePath = config.getString("KROKI_PLANTUML_INCLUDE_PATH");
+    if (plantUmlIncludePath != null) {
+      // NOTE: relying on classloading sequencing to ensure this gets set before ImportedFiles class is loaded by JVM
+      System.setProperty("plantuml.include.path", plantUmlIncludePath);
+    }
+  }
+
+  private static List<Pattern> parseIncludeWhitelist(String filename) {
+    if (filename != null && new File(filename).isFile()) {
+      try {
+        return Files.lines(Paths.get(filename)).filter(s -> !s.isEmpty()).map(Pattern::compile)
+          .collect(Collectors.toList());
+      } catch (IOException e) {
+        // ignore
```

Comment:
I think we should log the error with the context, so the user knows what is going on:

```
logger.warn("Unable to read the PlantUML whitelist file: xyz. The include whitelist will be empty.", e)
```